RoddyCore/src/de/dkfz/roddy/plugins/LibrariesFactory.groovy

```diff
@@ -531,10 +559,12 @@ public class LibrariesFactory extends Initializable {
             // version of it which may either be a revision (x:x.y-[0..n] or a higher compatible version.
             // Search the last valid entry in the chain.
             if (!usedPlugins.contains("${id}:${version}")) {
-                for (; pInfo.nextInChain != null; pInfo = pInfo.nextInChain) {
+                while (true) {
```

Comment:
It is actually not clear from the code here, in what order plugins are chained. I guess it is newest to oldest.

You could make a function `PlugInfo findNewestPluginInChain(PluginInfo)` that would be easy to test.